kubespawner/spawner.py

```diff
@@ -1757,6 +1783,82 @@ def start(self):
 
     _last_event = None
 
+    @gen.coroutine
+    def _make_create_pod_request(self, pod, request_timeout):
+        """
+        Make an HTTP request to create the given pod
+
+        Designed to be used with exponential_backoff, so returns
+        True / False on success / failure
+        """
```

Comment:
The function also relies on raising errors to signal an outcome. I'm not confident on how JupyterHub which is using a KubeSpawner object throwing an error will react. @yuvipanda do you know?

To have this docstring or similar describe a bit about that would be relevant for me at least, I've asked this myself many times while inspecting this code base and failed to spot the associated JupyterHub try/except logic as I assume this wouldn't bring the hub down.

Reply:
Letting unhandled errors raise is normal error behavior, i.e. spawn failed with an error, resulting in logged exception and 500 error. Using exponential backoff has three cases:

1. success (return True/truthy)
2. handled failure that should be retried (return False)
3. unhandled error (let it raise, will propagate up to the caller)

I don't think this PR changes the outcomes of different errors other than these short timeouts.